waiter/src/waiter/authorization.clj

```diff
@@ -32,6 +32,23 @@
   [entitlement-manager auth-user service-id service-description]
   (authorized? entitlement-manager auth-user :manage (make-service-resource service-id service-description)))
 
+(defn- make-token-resource
+  "Creates a resource from a token and token metadata for use with an entitlement manager"
+  [token {:strs [owner]}]
+  {:resource-type :token
+   :token token
+   :user owner})
+
+(defn manage-token?
+  "Returns whether the auth-user is allowed to modify the specified token."
+  [entitlement-manager auth-user token token-metadata]
+  (authorized? entitlement-manager auth-user :manage (make-token-resource token token-metadata)))
+
+(defn sync-token?
+  "Returns whether the auth-user is allowed to sync the specified token."
+  [entitlement-manager auth-user token token-metadata]
+  (authorized? entitlement-manager auth-user :sync (make-token-resource token token-metadata)))
```

Comment:
`:system` or `:admin` is a better name than `:sync`, leaving the option later to use this functionality for something else without the name being awkward.